Adicionado valor mínimo para saque

joaojvds opened the pull request:
-- Implementado texto informando qual é o limite mínimo de saque
-- Implementado check para impedir valor de saque abaixo do limite do fee
-- Implementado limite de saque na CurrencyApi
-- Implementado validação no mongoose para chegar se o amount é positivo

joaojvds committed:
Adicionado um limite minimo de saque no front

--Adicionado texto mostrando qual é o limite minimo de saque
--Adicionado check para impedir valor de saque abaixo do limite do fee

joaojvds committed:
Minor changes

joaojvds committed:
work in progress

joaojvds committed:
work in progress - Precisa ser testado

--A principio parece esta funcionando, é necessário mais testes

GCastilho committed:
CurrencyApi não aceita mais saque menor q 2*fee

-- Adicionado um check na CurrencyApi para bloquear requests de saque menores que 2*fee daquela currency
-- Adicionado um validador no schema das transactions para não permitir amount menor ou igual a zero
-- Reprojetado o beforeAll dos testes da /transactions para eles levarem o fee mínimo em consideração

GCastilho committed:
Adicionado check de amount no módulo externo

-- Adicionado um check para amount negativo em uma transação de saque ou recebimento nos módulos externos

joaojvds committed:
Merge branch 'master' into minimum-withdraw

joaojvds committed:
Descomentado check no front

GCastilho commented:
Precisa fazer os testes, eu não sei quando vou ter tempo, Quer se aventurar e começar eles, @joaojvds? Posso ir te guiando no _que_ precisa ser testado

joaojvds committed:
Implementado tests para o saque minimo

--Criado tests para testar valores abaixo do permitido
--minor changes

GCastilho commented on tests/CurrencyApi/operations.test.ts:
```diff
@@ -47,6 +47,12 @@ describe('Testing operations on the currencyApi', () => {
+				await expect(CurrencyApi.withdraw(user, currency, account, (fee * -0.01))).to.eventually.be
+					.rejectedWith(`Withdraw amount for ${currency} must be at least '${2 * fee}', but got ${fee * -0.01}`)
+			})
+
```
Coloca esse teste no final do arquivo, pq ele foi adicionado por último

GCastilho commented on tests/CurrencyApi/operations.test.ts:
```diff
@@ -60,23 +66,23 @@ describe('Testing operations on the currencyApi', () => {
-				 */
+					* Checa se houve erros de arredondamento comparando as
+					* as casas decimais significativas
+					*/
```
Corrige a indentação desse comment

GCastilho commented on tests/CurrencyApi/operations.test.ts:
```diff
@@ -60,23 +66,23 @@ describe('Testing operations on the currencyApi', () => {
-				 * Checa se houve erros no arredondamento somando os valores
-				 */
+					* Checa se houve erros no arredondamento somando os valores
+					*/
```
Desse tbm

GCastilho commented on client/src/routes/balances/_tableRow/withdrawCell.svelte:
```diff
@@ -45,7 +53,6 @@
 		margin: auto;
 		margin-top: 15px;
 		margin-bottom: 15px;
-		text-align: right;
```
Isso daqui tira o botão de withdraw do lado direito. O botão de withdraw deve ficar do lado direito com o texto

joaojvds replied:
se eu usar usar essa linha TODO texto sera alinhado para a direita incluindo o texto de erro, que eu quero que seja centralizado, mas eu já alterei o css para que o botão fique onde você quer.

GCastilho replied:
Coloca o texto do erro centralizado, ué. "Todo o form está para a direita exceto esse erro"
É pra isso que serve o "Cascade" do CSS

joaojvds replied:
e você acha que eu não tentei alinhar somente o texto do erro?

GCastilho replied:
Tu ta dizendo que não tem como dar override nisso?

GCastilho replied:
Aliás, eu acabei de ver aqui. Não ficou ruim a mensagem de erro do lado direito. Vc REALMENTE acha necessário que ela seja centralizada?
Eu acho que se encaixa ela estar à direita

GCastilho commented on client/src/routes/balances/_tableRow/withdrawCell.svelte:
```diff
@@ -90,28 +114,33 @@
 
 	p {
-		margin: 0
+		margin: 0;
+		text-align: right
```
Não há a necessidade desse p ter text-align pois o form terá o text-align para a direita

GCastilho reviewed (changes requested):
Vc não mudou o botão de lugar como eu disse
e quebrou os comentários do teste

joaojvds committed:
Minor fix

--Alterado alguns trechos no css para o botão ficar a direita
--Consertado comentários não indentados

GCastilho commented on client/src/routes/balances/_tableRow/withdrawCell.svelte:
```diff
@@ -42,18 +50,22 @@
 	}
 
 	form {
+		display: flex;
+		flex-direction: column;
```
Explica pra mim, pq vc teve que botar um flex aqui?

joaojvds replied:
eu não estava conseguindo mudar a posição do botão pelo display block, então eu mudei para um mais livre de eu alterar

GCastilho replied:
Entendi. Sem problemas
Mas faz do jeito q eu falei por whats, deixa o forma com text-align para a direita e _se quiser_ centraliza a msg de erro com display block e text-align

GCastilho commented on client/src/routes/balances/_tableRow/withdrawCell.svelte:
```diff
@@ -42,18 +50,22 @@
 	button {
 		border: 0;
 		border-radius: 10px;
+		width: 85.5625px;
+		height: 45px;
```
Isso n tava aqui antes, pq vc mudou o tamanho do botão?

joaojvds replied:
Eu preferi deixar o tamanho do botão setado para evitar que se deforme em alguma outra resolução, fora que esse botão é para ter um tamanho fixo

GCastilho replied:
Tem como a resolução dele deformar assim? Eita

GCastilho commented on client/src/routes/balances/_tableRow/withdrawCell.svelte:
```diff
@@ -42,18 +50,22 @@
+		height: 45px;
 		padding: 15px;
 		cursor: pointer;
+		align-self: flex-end;
```
Overengineering no CSS do botão

joaojvds replied:
ele esta se alinhando para a direita e é um comando bem comum, eu simplesmente usei uma propriedade do flexbox se mudar para direita também vai

GCastilho replied:
Saquei

GCastilho reviewed (changes requested):
Tira o flex do form, alinha para a direita e a mensagem de erro vc centraliza. Keep Things Simple

joaojvds committed:
Minor change

GCastilho reviewed (approved):
Pode fazer o commit, **usa a mensagem do PR** (eu editei ela)
